front/pages/w/[wId]/a/[aId]/index.tsx

```diff
@@ -188,61 +185,61 @@ export default function AppView({
     }, 1000);
   };
 
-  const update = (s: SpecificationType) => {
-    let c = extractConfig(s);
+  const update = async (s: SpecificationType) => {
+    const c = extractConfig(s);
     setRunnable(isRunnable(readOnly, s, c));
     setSpec(s);
     setConfig(c);
-    saveState(s, c);
+    await saveState(s, c);
   };
 
-  const handleNewBlock = (
+  const handleNewBlock = async (
     idx: number | null,
     blockType: BlockType | "map_reduce" | "while_end"
   ) => {
-    let s = addBlock(spec, idx === null ? spec.length - 1 : idx, blockType);
-    update(s);
+    const s = addBlock(spec, idx === null ? spec.length - 1 : idx, blockType);
+    await update(s);
     if (idx === null) {
       bottomRef.current?.scrollIntoView({ behavior: "smooth" });
     }
   };
 
-  const handleDeleteBlock = (idx: number) => {
-    let s = deleteBlock(spec, idx);
-    update(s);
+  const handleDeleteBlock = async (idx: number) => {
+    const s = deleteBlock(spec, idx);
+    await update(s);
```

Comment:
Same for these, I don't think these do anything as `handleDeleteBlock` is not awaited

Reply:
I think that's fine no?